libpod/runtime_ctr.go

```diff
@@ -547,6 +547,29 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 		toLock.lock.Lock()
 		defer toLock.lock.Unlock()
 	}
+	if ctr.config.Spec.Linux != nil &&
+	        (ctr.config.Spec.Linux.MaskedPaths != nil) &&
+		MountExists(ctr.config.Spec.Mounts, sysDevBlock) {
+		dir := fmt.Sprintf("%s/sysdevblock", ctr.state.RunDir)
+		for i, m := range ctr.config.Spec.Mounts {
+                        if m.Destination == sysDevBlock {
+				//ctr.config.Spec.Mounts[i].Source = dir
+				ctr.config.Spec.Mounts[i] = ctr.config.Spec.Mounts[len(ctr.config.Spec.Mounts)-1]
+				ctr.config.Spec.Mounts = ctr.config.Spec.Mounts[:len(ctr.config.Spec.Mounts)-1]
+                        }
+                }
+		ctr.state.BindMounts[sysDevBlock] = dir //  ??????????????
```

Comment:
In addition to removing comments here and formatting:

Is moving from a `Spec` to a `BindMount` here the right way?

Reply:
@mheon PTAL

Reply:
This is not safe. It may work for the first time the container is run, but it will be cleared for subsequent runs. Bind mounts are regenerated each time the container is run - they're only meant to handle files Podman creates and manages itself. Why do you want to use them here?

Reply:
This bind mount is for the symlinks that Podman creates itself. This seemed like a good location for a few symlinks derived from the device and volume mounts passed from the user in a place which I assume is auto-cleaned up.
```
$ ls -la /run/user/1000/containers/overlay-containers/cc8ca97825f48662209987fc11c34ebbea6a7c7a5ce39dde9a17cff036794b5b/userdata/sysdevblock
total 0
drwx------. 2 dan dan 100 Jul 14 17:09 .
drwx------. 4 dan dan 220 Jul 14 17:09 ..
lrwxrwxrwx. 1 dan dan  32 Jul 14 17:09 253:3 -> ../../devices/virtual/block/dm-3
lrwxrwxrwx. 1 dan dan  33 Jul 14 17:09 7:1 -> ../../devices/virtual/block/loop1
lrwxrwxrwx. 1 dan dan  33 Jul 14 17:09 7:2 -> ../../devices/virtual/block/loop2
```

I'm effectively using the `blkMnt` created in `SpecGenToOCI` (where visibility of Mask/unmask options exists) as a communication mechanism (was there another better var available?) to here, `setupContainer`, where `ctr.state.RunDir` exists so I can create the above dir, and `makeBindMounts` to mount it like resolv.conf/hosts etc (maybe `c.bindMountRootFile` should have been used).